pkg/component/networking/istio/charts/istio/istio-ingress/templates/vpn-envoy-filter.yaml

```
@@ -28,11 +28,20 @@ spec:
               routes:
               - match:
                   connect_matcher: {}
+                  headers:
+                    - name: Reversed-VPN
+                      string_match:
+                        safe_regex:
+                          regex: '^outbound\|1194\|\|vpn-seed-server(-[0-4])?\..*\.svc\.cluster\.local$'
                 route:
                   cluster_header: Reversed-VPN
                   upgrade_configs:
                   - connect_config: {}
                     upgrade_type: CONNECT
+              - match:
+                  connect_matcher: {}
+                direct_response:
+                  status: 403
```

Why not use the same approach as below and redirect to `https`?

```suggestion
                redirect:
                  https_redirect: true
                  port_redirect: 443
```

---

I thought it would be better to give clients using the correct connection method (“CONNECT”) more feedback about their request than the simple redirect.
Also, the previous ext_authz filter implementation also returned a 403 if the header was incorrect.

---

Well, you broadened the domain check thereby allowing all requests through. I would prefer if all failing requests be it legitimate or not being redirected to the https port. In other words, random connect queries should end at the https port as before.

---

Okay, accepted your suggestion :)